src/connectors/youtube.js

```diff
@@ -229,9 +257,7 @@ function getTrackInfoFromDescription() {
 }
 
 function getTrackInfoFromChapters() {
-	return Util.splitArtistTrack(
-		Util.getTextFromSelectors(chapterNameSelector)
-	);
+	return Util.splitArtistTrack(Util.getTextFromSelectors(chapterNameSelector), null, false, true);
```

Comment:
You should not use boolean type as a parameter. You cannot say what these `false` and `true` mean, unless you read documentation for this function.

That's why the `swap` argument is wrapped in an object, so you'll call it like
```js
Util.splitArtistTrack(
		Util.getTextFromSelectors(chapterNameSelector), null, { swap: true }
);
```

But I think this logic should not be implemented in this function. This is a quite rare case.

Reply:
> You should not use boolean type as a parameter. You cannot say what these `false` and `true` mean, unless you read documentation for this function.
> 
> That's why the `swap` argument is wrapped in an object, so you'll call it like
> 
> ```js
> Util.splitArtistTrack(
> 		Util.getTextFromSelectors(chapterNameSelector), null, { swap: true }
> );
> ```
> 
> But I think this logic should not be implemented in this function. This is a quite rare case.

Okay, I'll fix the boolean argument.

Do you have any suggestions for any other places where to implement the logic?  I think this function is the most appropriate one.  Besides, I've tried to preserve its interface and behavior as much as possible.

Reply:
Given that I have issues using this branch with [some album videos whose track names contain various separator characters](https://www.youtube.com/watch?v=aur4RKAgbqE), I'd like to ask if we should support artist names in chapter titles at all.  Admittedly, this is a really obscure corner case.  (What is more, it is entirely possible for a compilation video to contain tracks from different artists.)

Reply:
> Do you have any suggestions for any other places where to implement the logic?

You can use it in the `getTrackInfoFromChapters` functions. Just return the `track` property only. There's no sense to add this simple logic in other functions.

Reply:
By the way, when I suggested to wrap a boolean argument into an object, I meant wrapping multiple arguments in a *single* object.

```js
function test(arg1, { arg2 = false, arg3 = true } = {}) {
}

test(0, { arg2: true, arg3: false });
```

This is just a note, I'm still against adding the `allowTrackOnly` argument. :)

Reply:
> > Do you have any suggestions for any other places where to implement the logic?
> 
> You can use it in the `getTrackInfoFromChapters` functions. Just return an object with the `track` property only, if `Util.splitArtistTrack` returns an empty result.
> 
> There's no sense to add this simple logic in other functions.

Ok, I tend to agree that it would be an unnecessary complication of the utility functions, so I implemented it the way you want :)

> By the way, when I suggested to wrap a boolean argument into an object, I meant wrapping multiple arguments in a _single_ object.
Ah, *now* I understand the motivation behind your idea.  So the object would become something like a flags argument.